ONSdigital/eq-survey-runner
# Server Side Storage for Save and Resume

### Changes

This changes enables server side storage. The flask session will only hold the user id. All session state is now stored server side (e.g in a database). 

To enable this I've had to break the users questionnaire data from the users flask session, so it's required a massive refactor (sorry for the huge PR)
### How to test
1. Check out this branch
2. Enable database storage by setting an ENV setting `export EQ_SERVER_SIDE_STORAGE_TYPE=DATABASE`
3. Start completing a survey, make sure you submit the questionnaire page at least one to save the data
4. Kill the server
5. Restart and generate a new token
6. Check the survey restarts at the point you left off
### Who can test

Anyone apart from @warrenbailey
